lib/providers/quoters/RoutingApiQuoter.ts

```diff
@@ -22,12 +30,76 @@ export class RoutingApiQuoter implements Quoter {
 
     metrics.putMetric(`RoutingApiQuoterRequest`, 1);
     try {
-      const req = this.buildRequest(request);
+      let [resolvedTokenIn, resolveTokenOut]: [Currency | undefined, Currency | undefined] = [undefined, undefined];
+
+      try {
+        // we will need to call token fetcher to resolve the tokenIn and tokenOut
+        // there's no guarantee that the tokenIn and tokenOut are in the token address
+        // also the tokenIn and tokenOut can be native token
+        // portion service only accepts wrapped token address
+        [resolvedTokenIn, resolveTokenOut] = await Promise.all([
+          this.tokenFetcher.resolveTokenBySymbolOrAddress(request.info.tokenInChainId, request.info.tokenIn),
+          this.tokenFetcher.resolveTokenBySymbolOrAddress(request.info.tokenOutChainId, request.info.tokenOut),
+        ]);
+      } catch (e) {
+        // token fetcher can throw ValidationError,
+        // we must swallow the error and continue, meanwhile logging them,
+        // not being able to resolve the tokens here means we don't have portion amount
+        // throughout the lifecycle of this quote request processing,
+        // and we simply don't account for portion and let quote request processing continue
+        log.error({ e }, 'Failed to resolve tokenIn & tokenOut');
+        metrics.putMetric(`PortionProvider.resolveTokenErr`, 1);
```

Comment:
Nit: on naming convention. Might also want to also create an alarm for this? can be done in follow up
```suggestion
        metrics.putMetric('PortionProviderResolveTokenError', 1);
```